src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5522,7 +5522,7 @@ private sealed class VariableInfo
             internal Type LastAssignedType;
         }
 
-        private sealed class FindVariablesVisitor : AstVisitor
+        private sealed class FindVariablesVisitor : AstVisitor2
```

Comment:
Why do we need the AstVisitor2?

Reply:
Because do while/until loops use PipelineBaseAst as the condition and the new ChainableAst is based off that. Without this, the visitor would skip the ChainableAst and go straight to the PipelineAst and the parent wouldn't be the do loop.